### ISSUE TITLE
laptop keyboard issues

### ISSUE THREAD

**Issue by deepcdiver** [8bitbubsy/pt2-clone#8]
pressing return on its own is acting as pressing returns numpad.
not using fn.
USB keyboard works as intended.

**Comment by deepcdiver** [comment]
also would like to add that in order to enable unlock(fn+numlk) doesn't work while you're focusing on protracker.

**Comment by 8bitbubsy** [comment]
I can't really fix the problem because it works fine on my laptops, and I have no laptop that behaves like this...

**Comment by deepcdiver** [comment]
hi bubsy thanks for your attention. i dunno honestly I only found this issue today since I've been working a lot of protracker and the return key isn't working as intended , I plugged an USB keyboard and it works fine so I thought you could help me out with this.

**Comment by deepcdiver** [comment]
I've also tested your fast tracker 2 clone and the keys work fine so there's deffo an issue with protracker.

**Comment by deepcdiver** [comment]
tried that but no avail.

**Comment by 8bitbubsy** [comment]
Then I'm all out of ideas... :(

**Comment by deepcdiver** [comment]
yeah ano bro its really sad i stumbled upon this i dont know either if its my laptop or its the protracker keyboard emulation or something because ft2 clones shortcuts work fine and it gets the right messages off my keyboard. I'm as sad as you are believe me , I love protracker and the sound is really unique , me having to ditch it by not having to work with it directly it kinda sucks by having to resort to a different tracker to work the mods and then port them to pt2.

**Comment by 8bitbubsy** [comment]
I don't know. It might be because of a hook routine I am using on numlock (and the Windows key) to take over its functionality. Maybe the enter key on your keyboard depends on the numlock state, and that the numlock state is always set to one specific state when the PT2 clone is in focus.

**Comment by deepcdiver** [comment]
yeah and the thing is that the numlock state is always set to on

**Comment by 8bitbubsy** [comment]
On my PC, if numlock is off, it remains off when I start the PT2 clone... Just tested it. Kinda odd that it would remain on, on your laptop.

**Comment by deepcdiver** [comment]
yeah i just tried it again , no matter how  have numlk set by pressing fn and numlk its always set to on. 
whyyyyyyyyyyyyyyyyyyyyyyyyy?? XD

**Comment by deepcdiver** [comment]
what about if there was a way to change the keymaps would that solve it?

**Comment by deepcdiver** [comment]
so far i think the only issue is with the enter key everything else seems to do what's intended to do , its just that bloody numlock argh XD

**Comment by deepcdiver** [comment]
well i'm just gonna get another tracker to make mods and port them to pt2 or use ft2 or something. if there's nothing you can least can't say i haven't tried you know. maybe sometime you'll have the chance to stumble upon this same issue and you'll have the chance to cover this up. thanks bubsy.
